src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEnumerator.Unix.cs

```diff
@@ -87,8 +83,8 @@ private SafeDirectoryHandle CreateDirectoryHandle(string path)
 
         private void CloseDirectoryHandle()
         {
-            _directoryHandle?.Dispose();
-            _directoryHandle = null;
+            Interop.Sys.CloseDir(_directoryHandle);
```

Comment:
Seems like this should be guarded to only make the call if the ptr isn't null.

Reply:
Also if you are being particularly careful, Debug.Assert the return value of CloseDir is 0.

Reply:
I didn't think it was necessary, but I was on the fence- I'll add it

Reply:
In fact, perhaps tihs and CloseHandle should be wrapped in their Interop classes to at least Debug.Assert they succeed.

Reply:
I'm hesitant to add the assert- we never failed with SafeHandle.

Reply:
Looks like SafeHandle used to do that for Unix.

Reply:
SafeDirectoryHandle? Where is the assert?

Reply:
I was looking at SafefileHandle.Unix.cs:
```C#
            int result = Interop.Sys.Close(handle);
#if DEBUG
            if (result != 0)
            {
                Debug.Fail(string.Format(
                    "Close failed with result {0} and error {1}", 
                    result, Interop.Sys.GetLastErrorInfo()));
            }
#endif
```